ocs_ci/ocs/resources/objectbucket.py

```diff
@@ -108,126 +199,172 @@ def internal_verify_health(self):
         """
         raise NotImplementedError()
 
+    @abstractmethod
+    def internal_verify_deletion(self):
+        """
+        Abstract deletion verification method
+
+        """
+        raise NotImplementedError()
+
 
-class S3Bucket(MCGBucket):
+class MCG_CLIBucket(ObjectBucket):
     """
-    Implementation of an MCG bucket using the S3 API
+    Implementation of an MCG bucket using the NooBaa CLI
     """
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.mcg.s3_resource.create_bucket(Bucket=self.name)
+        self.mcg.exec_mcg_cmd(f'obc create --exact {self.name}')
```

Comment:
do we always create a bucket upon class instantiation?

Reply:
Yep. The classes in the file aren't meant to represent the OBC resources themselves - only to create and manipulate them. For representation of the associated data we use OBC(bucketname).

We can include a `self.obc` attribute that'll be available to OCBuckets and CLIBuckets, but it means it'll just take more time.
Since all of our tests already rely on having a separate variable holding this information, we won't shorten our code by accessing it via ObjectBucket.obc (as opposed to instantiating a new OBC(bucketname) object). It will cause longer runtimes, however, because it'll have to perform all the necessary `get()`s for each and every OBC created - even the ones whose OBC attributes we do not use.

Reply:
Discussion continued [here](https://github.com/red-hat-storage/ocs-ci/pull/2322#discussion_r460539124)